### PR TITLE
Bug 1872251: Included vendor verification

### PR DESCRIPTION
**Pull request by huffmanca**
**What is this PR about? / Why do we need it?**
Modifies the Makefile to allow a `make verify-deps` command which checks the go.sum and go.mod.

**What testing is done?** 
Local testing. The script produces output either way, but if the current go.sum/go.mod is different then it errors and reports the differences.

Note that https://github.com/openshift/release/pull/11615 adds the job to ci.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -23,6 +23,17 @@ GOPROXY=direct
...
 GOBIN=$(shell pwd)/bin
+GO ?=go
+deps_diff := diff --no-dereference -N
+
+# $1 - temporary directory
+define restore-deps
+	ln -s $(abspath ./) "$(1)"/current
+	cp -R -H ./ "$(1)"/updated
+	$(RM) -r "$(1)"/updated/vendor
+	cd "$(1)"/updated && $(GO) mod tidy && $(GO) mod vendor && $(GO) mod verify
```
Comment (gnufied) [inline Makefile:34]: shouldn't we run `go mod tidy`  after `go mod vendor`?

File: Makefile [hunk 2]
```diff
@@ -23,6 +23,17 @@ GOPROXY=direct
...
 GOOS=$(shell go env GOOS)
 GOBIN=$(shell pwd)/bin
+GO ?=go
+deps_diff := diff --no-dereference -N
+
+# $1 - temporary directory
+define restore-deps
+	ln -s $(abspath ./) "$(1)"/current
+	cp -R -H ./ "$(1)"/updated
+	$(RM) -r "$(1)"/updated/vendor
```
Comment (gnufied) [inline Makefile:33]: is there a reason to use variable for "rm" ?

File: Makefile (hunk 1 above)
Comment (huffmanca) [reply Makefile]: Adjusted this to be after `go mod vendor`.

File: Makefile (hunk 2 above)
Comment (huffmanca) [reply Makefile]: Not particularly. I went ahead and removed this, along with adding the `-f` flag to rm.

**Pull request merged**